integrationTests/sync/common.go

```diff
@@ -0,0 +1,182 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+var stepDelay = time.Second
+var delayP2pBootstrap = time.Second * 2
+var stepSync = time.Second * 2
+
+func setupSyncNodesOneShardAndMeta(
+	numNodesPerShard int,
+	numNodesMeta int) ([]*integrationTests.TestProcessorNode, p2p.Messenger, []int) {
+
+	maxShards := uint32(1)
+	shardId := uint32(0)
+
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+	advertiserAddr := integrationTests.GetConnectableAddress(advertiser)
+
+	nodes := make([]*integrationTests.TestProcessorNode, 0)
+	for i := 0; i < numNodesPerShard; i++ {
+		shardNode := integrationTests.NewTestSyncNode(
+			maxShards,
+			shardId,
+			shardId,
+			advertiserAddr,
+		)
+		nodes = append(nodes, shardNode)
+	}
+	idxProposerShard0 := 0
+
+	for i := 0; i < numNodesMeta; i++ {
+		metaNode := integrationTests.NewTestSyncNode(
+			maxShards,
+			sharding.MetachainShardId,
+			shardId,
+			advertiserAddr,
+		)
+		nodes = append(nodes, metaNode)
+	}
+	idxProposerMeta := numNodesPerShard
```

Comment:
it is confusing this initialization. make it more generic - like making the last node from the list the metachain node proposer.

Reply:
Made the last node the proposer